# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1333512989 25200
#      Tue Apr 03 21:16:29 2012 -0700
# Node ID f739e30494c8b7f700e520c7ed190c67ba175459
# Parent  3b19a952ab54292ba0bf8f883d7fdd5b56605786
condest.m: Fix failing %!test.

* condest.m: Use is_function_handle rather than isscalar to determine
if input is a function handle.

diff --git a/scripts/linear-algebra/condest.m b/scripts/linear-algebra/condest.m
--- a/scripts/linear-algebra/condest.m
+++ b/scripts/linear-algebra/condest.m
@@ -127,17 +127,17 @@ function [est, v] = condest (varargin)
     A = varargin{1};
     if (! issquare (A))
       error ("condest: matrix must be square");
     endif
     n = rows (A);
     have_A = true;
 
     if (nargin > 1)
-      if (isscalar (varargin{2}))
+      if (! is_function_handle (varargin{2}))
         t = varargin{2};
         have_t = true;
       elseif (nargin > 2)
         solve = varargin{2};
         solve_t = varargin{3};
         have_solve = true;
         if (nargin > 3)
           t = varargin{4};
